packages/disclosure/src/index.tsx

```diff
@@ -7,16 +7,10 @@ export interface DisclosureSharedProps {
     className?: string;
 }
 
-export type RegisterDetails = () => {
-    handleClick: (e: React.MouseEvent) => void;
-    index: number;
-};
-
 export interface DisclosureRootProps {
-    children: (registerDetails: RegisterDetails) => React.ReactNode | React.ReactNode[];
+    children: React.ReactNode[];
```

Comment:
I don't think this does what you think it does. What are you trying to do here? `ReactNode` (without the `[]`) can be a single node or an array of nodes or some other stuff, you don't need to explicitly make it an array.

Reply:
I get it! Thanks!